tests/unit/common.py

```diff
@@ -317,6 +317,15 @@ def __call__(self, request):
         else:
             world_size = self.world_size
 
+        if isinstance(world_size, int):
+            max_world_size = world_size
+        else:
+            max_world_size = max(world_size)
+        if torch.cuda.device_count() < max_world_size:
+            pytest.skip(
+                f"Skipping test because not enough GPUs are available: {max_world_size} required, {torch.cuda.device_count()} available"
+            )
```

Comment:
I think this check needs to happen in the `DistributedExec` class or we may still run into problems with `DistributedFixture`. Also, we still want to run a test that has multiple world sizes if possible. With the current changes, a test that has world sizes of [2,4] would get skipped altogether on a machine with 2 GPUs.

I suggest adding the following lines:
```python
if torch.cuda.device_count() < num_procs:
    pytest.skip(f"Skipping test because not enough GPUs are available: {num_procs} required, {torch.cuda.device_count()} available")
```
To this method:
https://github.com/microsoft/DeepSpeed/blob/b6030a913b8cb14aa5f9529d8d7854cd15acefc0/tests/unit/common.py#L107